carball/json_parser/game.py

```diff
@@ -304,17 +304,64 @@ def parse_replay(self):
                             unique_id = str(
                                 actor_data['Engine.PlayerReplicationInfo:UniqueId']['unique_id']['remote_id'][
                                     actor_type])
-                            leader_actor_type = list(
-                                actor_data["TAGame.PRI_TA:PartyLeader"]["party_leader"]["id"][0].keys()
-                            )[0]
-                            leader = str(
-                                actor_data["TAGame.PRI_TA:PartyLeader"]["party_leader"]["id"][0][leader_actor_type]
-                            )
-                            if leader in parties:
-                                if unique_id not in parties[leader]:
-                                    parties[leader].append(unique_id)
-                            else:
-                                parties[leader] = [unique_id]
+
+                            # only process if party_leader id exists
+                            if "party_leader" in actor_data["TAGame.PRI_TA:PartyLeader"] and \
```

Comment:
If all b this code is just for party handling do you think it would be possible to put it into its own method?

It's a good time to start cleaning this up a bit.